booster/library/Booster/Pattern/Rewrite.hs

```diff
@@ -359,7 +359,8 @@ applyRule pat@Pattern{ceilConditions} rule = withRuleContext rule $ runRewriteRu
             checkAllRequires <- SMT.checkPredicates solver prior mempty (Set.fromList unclearRequires)
 
             case checkAllRequires of
-                Nothing -> do
+                Left _ -> do
+                    -- TODO: we could process SMTError here
```

Comment:
We should probably restrict this to the `SMTSolverUnknown` case. An `SMTTranslationError` is an entirely different case (and the `General` error was meant to cover infrastructure issues).